IdentityCore/src/webview/systemWebview/MSIDSystemWebviewController.m

```diff
@@ -26,19 +26,100 @@
 //------------------------------------------------------------------------------
 
 #import "MSIDSystemWebviewController.h"
+#import "MSIDSFAuthenticationSession.h"
+#import "MSIDSafariViewController.h"
+#import "MSIDWebviewAuthorization.h"
+#import "MSIDOauth2Factory.h"
 
 @implementation MSIDSystemWebviewController
+{
+    id<MSIDRequestContext> _context;
+
+    MSIDSFAuthenticationSession *_authSession;
+    MSIDSafariViewController *_safariViewController;
+    
+    MSIDOauth2Factory *_factory;
+}
 
-@synthesize parentViewController;
+- (instancetype)initWithStartURL:(NSURL *)startURL
+               callbackURLScheme:(NSString *)callbackURLScheme
+                         context:(id<MSIDRequestContext>)context
+{
+    self = [super init];
+    
+    if (self)
+    {
+        _startURL = startURL;
+        _context = context;
+        _callbackURLScheme = callbackURLScheme;
+    }
+    
+    return self;
+}
 
-- (void)startRequestWithCompletionHandler:(MSIDWebUICompletionHandler)completionHandler
+- (void)startWithCompletionHandler:(MSIDWebUICompletionHandler)completionHandler
 {
+    if (!_startURL)
+    {
+        NSError *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInvalidParameter, @"Attemped to start with nil URL", nil, nil, nil, _context.correlationId, nil);
+        completionHandler(nil, error);
+        return;
+    }
     
+    if (@available(iOS 11.0, *))
+    {
+        _authSession = [[MSIDSFAuthenticationSession alloc] initWithURL:self.startURL
+                                                      callbackURLScheme:self.callbackURLScheme
+                                                                context:_context];
+
+        if (_authSession)
+        {
+            [_authSession startWithCompletionHandler:completionHandler];
+            return;
+        }
+    }
+    else
+    {
+        _safariViewController = [[MSIDSafariViewController alloc] initWithURL:_startURL
+                                                                     context:_context];
+        
+        if (_safariViewController)
+        {
+            [_safariViewController startWithCompletionHandler:completionHandler];
+            return;
+        }
+    }
+    
+    NSError *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInteractiveSessionStartFailure, @"Failed to create an auth session", nil, nil, nil, _context.correlationId, nil);
+    completionHandler(nil, error);
 }
 
+
 - (void)cancel
 {
+    if (@available(iOS 11.0, *))
+        [_authSession cancel];
+    else
+        [_safariViewController cancel];
+}
+
+- (BOOL)handleURLResponseForSafariViewController:(NSURL *)url
+{
+    if (!url)
+    {
+        MSID_LOG_ERROR(_context, @"nil passed into the MSID Web handle response.");
+        return NO;
+    }
+    
+    if (@available(iOS 11.0, *)) { return NO; }
```

Comment:
nit: should this be yes as we don't have anything to handle?

Reply:
no, it should be no if we don't handle it.
think about it from this angle, 
it app delegate, it will do something like
return microsoft_url_handle || other_url_handle